src/Command/Key/Child.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Command.Key.Child
+    ( Cmd (..)
+    , mod
+    , run
+    ) where
+
+import Prelude hiding
+    ( mod )
+
+import Cardano.Address.Derivation
+    ( deriveXPrv, deriveXPub, xprvToBytes, xpubToBytes )
+import Codec.Binary.Bech32.TH
+    ( humanReadablePart )
+import Control.Monad
+    ( foldM )
+import Data.Functor.Identity
+    ( Identity (..) )
+import Options.Applicative
+    ( CommandFields, Mod, command, footerDoc, helper, info, progDesc )
+import Options.Applicative.Derivation
+    ( DerivationPath
+    , DerivationScheme
+    , castDerivationPath
+    , derivationPathArg
+    , derivationSchemeOpt
+    )
+import Options.Applicative.Encoding
+    ( Encoding, encodingOpt )
+import Options.Applicative.Help.Pretty
+    ( string )
+import System.IO
+    ( stdin, stdout )
+import System.IO.Extra
+    ( failWith, hGetXP__, hPutBytes )
+
+
+data Cmd = Child
+    { encoding :: Encoding
+    , scheme :: DerivationScheme
+    , path :: DerivationPath
+    } deriving (Show)
+
+mod :: (Cmd -> parent) -> Mod CommandFields parent
+mod liftCmd = command "child" $
+    info (helper <*> fmap liftCmd parser) $ mempty
+        <> progDesc "Derive child keys from a parent public/private key."
+        <> footerDoc (Just $ string $ mconcat
+            [ "The parent key is read from stdin."
+            ])
+  where
+    parser = Child
+        <$> encodingOpt [humanReadablePart|???|]
```

Comment:
what ??? exactly suppose to do here ?

Reply:
The `child` command works for both `xprv` and `xpub`. We only know what sort of key is given when handling the command. Note also that the "encoding" is only use for the _output_, not for decoding inputs (here we don't assume any encoding, and we try to decode any known encoding). 

What happens is that, I simply `fmap` over the encoding when the right prefix is known, to replace the `???` with an appropriate value:

https://github.com/input-output-hk/cardano-addresses/pull/29/files#diff-63fa7f406d9975760ab8aa7a21b2260dR77
```hs
                    pure
                        ( xpubToBytes child
                        , fmap (const [humanReadablePart|xpub|] ) encoding
                        )
```